Tests/MapboxDirectionsTests/RouteStepTests.swift

```diff
@@ -106,59 +106,125 @@ class RoadTests: XCTestCase {
+        let options = RouteOptions(coordinates: [CLLocationCoordinate2D(latitude: 0, longitude: 0), CLLocationCoordinate2D(latitude: 1, longitude: 1)])
+        options.shapeFormat = .polyline
         
-        let encodedData = NSMutableData()
-        let keyedArchiver = NSKeyedArchiver(forWritingWith: encodedData)
-        keyedArchiver.requiresSecureCoding = true
-        keyedArchiver.encode(step, forKey: "step")
-        keyedArchiver.finishEncoding()
+        let decoder = JSONDecoder()
+        decoder.userInfo[.options] = options
         
-        let keyedUnarchiver = NSKeyedUnarchiver(forReadingWith: encodedData as Data)
-        keyedUnarchiver.requiresSecureCoding = true
-        let unarchivedStep = keyedUnarchiver.decodeObject(of: RouteStep.self, forKey: "step")!
-        keyedUnarchiver.finishDecoding()
+        let step = try! decoder.decode(RouteStep.self, from: routeStepJSON.data(using: .utf8)!)
         
-        XCTAssertNotNil(unarchivedStep)
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.prettyPrinted]
+        let encoded = try! encoder.encode(step)
+        let roundTripJSON = String(data: encoded, encoding: .utf8)
         
-        XCTAssertEqual(unarchivedStep.coordinates?.count, step.coordinates?.count)
-        XCTAssertEqual(unarchivedStep.coordinates?.first?.latitude, step.coordinates?.first?.latitude)
-        XCTAssertEqual(unarchivedStep.coordinates?.first?.longitude, step.coordinates?.first?.longitude)
-        XCTAssertEqual(unarchivedStep.instructions, step.instructions)
-        XCTAssertEqual(unarchivedStep.initialHeading, step.initialHeading)
-        XCTAssertEqual(unarchivedStep.finalHeading, step.finalHeading)
-        XCTAssertEqual(unarchivedStep.maneuverType, step.maneuverType)
-        XCTAssertEqual(unarchivedStep.maneuverDirection, step.maneuverDirection)
-        XCTAssertEqual(unarchivedStep.maneuverLocation.latitude, step.maneuverLocation.latitude)
-        XCTAssertEqual(unarchivedStep.maneuverLocation.longitude, step.maneuverLocation.longitude)
-        XCTAssertEqual(unarchivedStep.exitIndex, step.exitIndex)
-        XCTAssertEqual(unarchivedStep.distance, step.distance)
-        XCTAssertEqual(unarchivedStep.expectedTravelTime, step.expectedTravelTime)
-        XCTAssertEqual(unarchivedStep.names ?? [], step.names ?? [])
-        XCTAssertEqual(unarchivedStep.phoneticNames ?? [], step.phoneticNames ?? [])
-        XCTAssertEqual(unarchivedStep.transportType, step.transportType)
-        XCTAssertEqual(unarchivedStep.destinations ?? [], step.destinations ?? [])
-        XCTAssertEqual(unarchivedStep.instructionsSpokenAlongStep ?? [], step.instructionsSpokenAlongStep ?? [])
-        XCTAssertEqual(unarchivedStep.instructionsDisplayedAlongStep ?? [], step.instructionsDisplayedAlongStep ?? [])
-        XCTAssertEqual(unarchivedStep.drivingSide, DrivingSide.left)
+        XCTAssert(roundTripJSON == pass)
+    }
+}
+
+
+fileprivate let routeStepJSON = """
```

Comment:
This massive multiline-stringified JSON source code should be reverted back to what we had before. The aim is to unit-test RouteStep, not integration-test nearly the entire library.